tfhe/src/core_crypto/gpu/vec.rs

```diff
@@ -66,6 +67,25 @@ impl<T: Numeric> CudaVec<T> {
     pub fn is_empty(&self) -> bool {
         self.len == 0
     }
+
+    pub(crate) fn range_bounds_to_start_end<R>(&self, range: R) -> (usize, usize)
+    where
+        R: std::ops::RangeBounds<usize>,
+    {
+        let start = match range.start_bound() {
+            Unbounded => 0usize,
+            Included(start) => *start,
+            Excluded(start) => *start + 1,
+        };
+
+        let end = match range.end_bound() {
+            Unbounded => self.len() - 1,
+            Included(end) => *end,
+            Excluded(end) => *end - 1,
+        };
+
+        (start, end)
```

Comment:
I'm going to nitpick on this:

Generally when given a range as in start, end end is exlusive.

Here you choose that end is inclusive, that's ok, but we should be explicit by making this function return an `std::ops::RangeInclusive` (return `start..=end`), or if you change it to be exclusive return `std::ops::Range` (start..end).

Reply:
Hmm actually looking at it in detail again I think what I did is consistent:
- this function takes a range (start..end) as argument and returns the two usize values for the start and end indexes
- range.end_bound() will return end - 1 if the end bound of the input range is exculsive, and end if it is inclusive: I think this is the expected behavior

Reply:
Thank you for the clarification on the phone :slightly_smiling_face: